src/client/texturesource.cpp

```diff
@@ -1848,58 +1856,115 @@ bool TextureSource::generateImagePart(std::string_view part_of_name,
 
 #undef CHECK_DIM
 
-/*
-	Calculate the color of a single pixel drawn on top of another pixel.
 
-	This is a little more complicated than just video::SColor::getInterpolated
-	because getInterpolated does not handle alpha correctly.  For example, a
-	pixel with alpha=64 drawn atop a pixel with alpha=128 should yield a
-	pixel with alpha=160, while getInterpolated would yield alpha=96.
-*/
-static inline video::SColor blitPixel(const video::SColor src_c, const video::SColor dst_c, u32 ratio)
+namespace {
+
+/// Draw src on top of dst
+template <bool overlay, class Colour>
+void blit_pixel(const Colour &src, Colour &dst)
```

Comment:
again: `sizeof(Colour) < sizeof(void*)`
though since this is a template the compiler will optimize that away anyway